onnxscript/function_libs/torch_aten/ops/core.py

```diff
@@ -269,10 +269,60 @@ def aten_angle(self: TensorType) -> TensorType:
     raise NotImplementedError()
 
 
-def aten_any(self: TensorType) -> TensorType:
+@torch_op("aten::any", trace_only=True)
+def aten_any(self: BOOL, dim: Optional[int] = None, keepdim: bool = True) -> BOOL:
     """any(Tensor self) -> Tensor"""
 
-    raise NotImplementedError()
+    zero = op.Constant(value_float=0.0)
+    minus_1 = op.Constant(value_ints=[-1])
+    self = op.Cast(self, to=FLOAT.dtype)
+    self_rank = op.Size(op.Shape(self))
+    if self_rank == 0:
+        self = op.Reshape(self, minus_1)
+
+    if op.OptionalHasElement(dim):
+        result = op.Greater(self, zero)
+        result_int = op.Cast(result, to=FLOAT.dtype)
+        dim = op.Reshape(dim, minus_1)
+        dims = op.Cast(dim, to=INT64.dtype)
+        result_sum = op.ReduceSum(result_int, dims, keepdims=keepdim, noop_with_empty_axes=0)
+        result = op.Greater(result_sum, zero)
+    else:  # kwarg{dim,keepdim} was not given
+        result = op.Greater(self, zero)
+        result_rank = op.Size(op.Shape(result))
+        if result_rank > 0:  # need return rank=0 result
+            result_int = op.Cast(result, to=FLOAT.dtype)
+            result_sum = op.ReduceSum(result_int, keepdims=0, noop_with_empty_axes=0)
+            result = op.Greater(result_sum, zero)
```

Comment:
I wonder if we can just cast everything to BOOL then reduce

Reply:
Hopefully only `0.0` will be casted to `0`?

Reply:
that means -0.0000001 or 0.00000001 stays `True`

Reply:
maybe something like `Not(Equal(0)) -> Cast(INT) -> ReduceMax -> Cast(BOOL)` 😅

Reply:
Or maybe this is better:

`Cast(BOOL) -> Cast(UINT8) -> ReduceMax -> Cast(BOOL)` so there is not equality and simulation of the casting.

Reply:
Yes, that was my suggestion. (There is a minor downside/implication, since ReduceMax switched from attribute to input, for axes, in opset 18 ... while ReduceSum has been uniformly so, since opset13. So, this will complicate the opsets that can be targetted potentially.)

Reply:
Do we worry that ReduceSum may overflow?